common/config/subspaces/default/.pnpmfile.cjs

```diff
@@ -62,6 +62,12 @@ function readPackage(packageJson, context) {
       fixUndeclaredDependency(packageJson, '@typescript-eslint/parser');
       break;
     }
+
+    case '@pnpm/lockfile-file': {
+      // The `@pnpm/lockfile-file` package requires `@pnpm/logger@^5.0.0`, but we are using `@pnpm/logger@1001.0.0`.
+      packageJson.peerDependencies['@pnpm/logger'] = '*';
+      break;
```

Comment:
@gavinxgu unlike the above rules, this new rule seems to affect the dependency graph of the published Rush package.  That would mean that the dependency versions being tested by our CI will be different from somebody who does `npm install -g @microsoft/rush`, right?  If so, is there a way to avoid this problem? (e.g. `rush update --full`)

Reply:
I've got an idea - similar to how Heft storybook kit does it, we can organize pnpm's internal packages by version into separate packages like rush-pnpm-kit-v8, rush-pnpm-kit-v9, and rush-pnpm-kit-v10. Then when we need to use different version-specific methods, we just import from the corresponding lib. This way we can avoid those unmet peerDeps issues.